File: app/domain/entities/states/pendiente_state.rb
require_relative './reserva_state'
require_relative '../../enums/estado_reserva' # Corrected path
require_relative './cancelada_state'
require_relative './confirmada_state'

module Entities
  module States
    class PendienteState < ReservaState
      def nombre
        Enums::EstadoReserva::PENDIENTE
      end

      def cancelar(reserva, ahora)
        # Lanza una excepción si la política de cancelación no permite la cancelación.
        unless reserva.politica_cancelacion.puede_cancelar(reserva.slot.inicio, ahora)
          raise StandardError, 'Cancelación fuera de plazo. Aplica cargo.'
        end

        reserva.estado_interno = CanceladaState.new
        reserva.slot.cancelar
      end

      def confirmar(reserva)
        reserva.estado_interno = ConfirmadaState.new
      end
    end
  end
end

File: EduLink.Domain/Entities/States/PendienteState.cs
using EduLink.Domain.Enums;
using System;

namespace EduLink.Domain.Entities.States;

public class PendienteState : ReservaState
{
	public override EstadoReserva Nombre => EstadoReserva.Pendiente; // Aquí se usa el enum pero no es necesario

	public override void Cancelar(Reserva reserva, DateTime ahora)
	{
		if (!reserva.PoliticaCancelacion.PuedeCancelar(reserva.Slot.Inicio, ahora))
			throw new InvalidOperationException("Cancelación fuera de plazo. Aplica cargo.");

		reserva.EstadoInterno = new CanceladaState();
		reserva.Slot.Cancelar();
	}

	public override void Confirmar(Reserva reserva)
	{
		reserva.EstadoInterno = new ConfirmadaState();
	}
}